app/models/miq_server/worker_management/monitor/stop.rb

```diff
@@ -41,11 +41,9 @@ def stop_worker(worker, monitor_reason = nil)
       w.stop_container
     elsif w.systemd_worker?
       w.stop_systemd_worker
-    elsif w.respond_to?(:terminate)
-      w.terminate
     else
       w.update(:status => MiqWorker::STATUS_STOPPING)
-      worker_set_message(w, 'exit')
+      Process.kill("TERM", w.pid)
```

Comment:
Do we need to check `w.pid.nil?`

Reply:
This will fail if the pid is nil. If we have a better solution for what to do if/when we have a nil pid then sure. Would we rather log an error and move on in that case?

Reply:
Yeah I think log and move on.